Include/Acidanthera/Library/OcVirtualFsLib.h

```diff
@@ -66,6 +94,82 @@ CreateRealFile (
   IN OUT EFI_FILE_PROTOCOL  **File
   );
 
+/**
+  Creates read-only EFI_FILE_PROTOCOL virtual directory instance,
+  optionally as an overlay over an existing EFI_FILE_PROTOCOL instance.
+  On success ownership of FileName and UnderlyingFile
+  is transferred to the resulting EFI_FILE_PROTOCOL, which frees them
+  with FreePool upon closing EFI_FILE_PROTOCOL.
+
+  Resulting EFI_FILE_PROTOCOL has 2nd revision, but may be downgraded
+  to 1st by updating the corresponding field.
+
+  @param[in]      FileName         Pointer to the file's name.
+  @param[in]      ModificationTime File modification date, optional.
+  @param[in]      UnderlyingFile   The underlying file protocol, optional.
+  @param[in,out]  File             Resulting file protocol.
+
+  @return  EFI_SUCCESS if instance was successfully created.
+**/
+EFI_STATUS
+VirtualDirCreateOverlay (
+  IN     CHAR16             *FileName,
+  IN     EFI_TIME           *ModificationTime OPTIONAL,
+  IN     EFI_FILE_PROTOCOL  *UnderlyingFile OPTIONAL,
+  IN OUT EFI_FILE_PROTOCOL  **File
+  );
+
+/**
+  Creates read-only EFI_FILE_PROTOCOL virtual directory instance,
+  optionally as an overlay over an existing EFI_FILE_PROTOCOL instance.
```

Comment:
I do not see `optionally as an overlay over an existing EFI_FILE_PROTOCOL instance` in the code. `File` is always overwritten regardless of the passed value. What do I miss?